packages/sanity/src/core/store/_legacy/document/document-pair/editState.ts

```diff
@@ -51,14 +55,15 @@ export const editState = memoize(
           ),
         ]),
       ),
-      map(([draftSnapshot, publishedSnapshot, transactionSyncLock]) => ({
+      swr(`${idPair.publishedId}-${idPair.draftId}`),
+      map(({value: [draftSnapshot, publishedSnapshot, transactionSyncLock], fromCache}) => ({
         id: idPair.publishedId,
         type: typeName,
         draft: draftSnapshot,
         published: publishedSnapshot,
         liveEdit,
-        ready: true,
-        transactionSyncLock,
+        ready: !fromCache,
```

Comment:
would it make sense to pass `fromCache` as a separate property on EditState? The granularity might be useful in certain cases (can do this later if the need arises though)